src/nci_util.c

```diff
@@ -183,6 +194,19 @@ nci_parse_mode_param(
             GDEBUG("  PollB.fsc = %u", ppb->fsc);
             GDEBUG("  PollB.nfcid0 = %02x %02x %02x %02x", ppb->nfcid0[0],
                 ppb->nfcid0[1], ppb->nfcid0[2], ppb->nfcid0[3]);
+            GDEBUG("  PollB.AppData = %02x %02x %02x %02x", ppb->app_data[0],
+                ppb->app_data[1], ppb->app_data[2], ppb->app_data[3]);
+
+            GString *buf = g_string_new(NULL);
```

Comment:
No declarations in the middle of the function, please.

Fortunately, there's no need to move it too far. This allocation/formatting stuff needs to be wrapped into `if (GLOG_ENABLED(GLOG_LEVEL_DEBUG)) { ... }`  block anyway. When debug logging if off (and it's off most of the time), there's no reason to allocate a buffer and format it, only to throw it away. It's a waste.

There are example in this file how to do it _the right way_ (tm).

Reply:
And don't forget `#if GUTIL_LOG_DEBUG` too. If debug logging is disabled at compile time, there's no need to even compile that code because in this case `GDEBUG` is no-op.